### PR TITLE
FSDK-99: Typescript ReactionList

### PR DESCRIPTION
**Pull request by mahboubii**

### CODE REVIEW

File: src/components/ReactionList.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        reverse: reverseOrder,
+        children: reactions.map((reaction) =>
+          smartRender(Reaction, {
+            reaction: reaction.toJS(),
+            key: reaction.get('id'),
+          }),
+        ),
+      })}
+    </>
+  );
```
Comment (arnautov-anton) [inline src/components/ReactionList.tsx:89]:
```suggestion
  return smartRender<LoadMorePaginatorProps>(Paginator, {
    loadNextPage,
    hasNextPage,
    refreshing,
    reverse: reverseOrder,
    children: reactions.map((reaction) =>
      smartRender(Reaction, {
        reaction: reaction.toJS(),
        key: reaction.get('id'),
      }),
    ),
  });
```

Comment (mahboubii) [reply src/components/ReactionList.tsx:89]: TS is not happy with the return value of smartRender

File: src/components/ReactionList.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    immutable.List(),
+  ) as immutable.List<immutable.Record<EnrichedReaction>>;
+  const refreshing = feed.activities.getIn(
+    [...activityPath, orderPrefix + '_reactions_extra', reactionKind, 'refreshing'],
+    false,
+  );
+
+  if (reverseOrder) {
+    reactions = reactions.reverse();
+  }
```
Comment (arnautov-anton) [inline src/components/ReactionList.tsx:62]:
```suggestion
  if (reverseOrder) reactions.reverse();
```

File: src/components/ReactionList.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  activityPath: defaultActivityPath,
+  oldestToNewest = false,
+  Paginator = LoadMorePaginator,
+  reverseOrder = false,
+}: ReactionListType<UT, RT, CRT>) => {
+  const feed = useFeedContext<UT, AT, CT, RT, CRT, PT>();
+
+  const activityPath = defaultActivityPath || feed.getActivityPath(activityId);
+  const orderPrefix = oldestToNewest ? 'oldest' : 'latest';
+  const reactions_extra = feed.activities.getIn([...activityPath, orderPrefix + '_reactions_extra']);
```
Comment (arnautov-anton) [inline src/components/ReactionList.tsx:49]:
```suggestion
  const reactionsExtra = feed.activities.getIn([...activityPath, orderPrefix + '_reactions_extra']);
```

File: src/components/ReactionList.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  if (reverseOrder) {
+    reactions = reactions.reverse();
+  }
+
+  const loadNextPage = () => {
+    feed.loadNextReactions(activityId, reactionKind, activityPath, oldestToNewest);
+  };
+
+  useEffect(() => {
+    if (oldestToNewest && reactions_extra) {
```
Comment (arnautov-anton) [inline src/components/ReactionList.tsx:69]:
```suggestion
    if (oldestToNewest && reactionsExtra) {
```

File: src/components/ReactionList.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  oldestToNewest = false,
+  Paginator = LoadMorePaginator,
+  reverseOrder = false,
+}: ReactionListType<UT, RT, CRT>) => {
+  const feed = useFeedContext<UT, AT, CT, RT, CRT, PT>();
+
+  const activityPath = defaultActivityPath || feed.getActivityPath(activityId);
+  const orderPrefix = oldestToNewest ? 'oldest' : 'latest';
+  const reactions_extra = feed.activities.getIn([...activityPath, orderPrefix + '_reactions_extra']);
+  const hasNextPage = reactions_extra ? !!reactions_extra.getIn([reactionKind, 'next'], '') : true;
```
Comment (arnautov-anton) [inline src/components/ReactionList.tsx:50]:
```suggestion
  const hasNextPage = !!reactions_extra?.getIn([reactionKind, 'next'], '') ?? true;
```

**Review by arnautov-anton — APPROVED** [review]

**Pull request merged**